sys/include/net/gnrc/sixlowpan/frag/rb.h

```diff
@@ -102,13 +102,17 @@ typedef struct {
  * @param[in] netif_hdr     The interface header of the fragment, with
  *                          gnrc_netif_hdr_t::if_pid and its source and
  *                          destination address set.
- * @param[in] frag          The fragment to add.
+ * @param[in] frag          The fragment to add. Will be released by the
+ *                          function.
  * @param[in] offset        The fragment's offset.
  * @param[in] page          Current 6Lo dispatch parsing page.
+ *
+ * @return  The reassembly buffer entry the fragment was added to on success.
+ * @return  NULL on error or when it was consumed by another 6LoWPAN layer.
```

Comment:
> […] or when it was consumed by another 6LoWPAN layer.

I also should rethink this... With the existing code it makes things easy, but we basically never get feedback on the first fragment any type of header compression when implemented this way...

Reply:
Ok. I just removed the call to `dispatch_when_complete()` in IPHC. This breaks a bit with the model proposed in https://github.com/RIOT-OS/RIOT/issues/8511, as the data flow of the reassembled packet now returns to the fragmentation layer. But I don't see any other way to do this and in the end, fragmentation should handle fragmentation stuff, so maybe for this one exception it is cleaner to break the model.

Reply:
Another advantage: I could now remove the NOP for `dispatch_when_complete()` as it will now only be used for sure if the reassembly buffer is compiled in. ;-).